### ISSUE TITLE
not found error, while cargo check & build works fine

### ISSUE THREAD

**Issue by Zercerium** [rust-lang/rust-analyzer#16552]
```rs
/Users/nick/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/test/src/test_result.rs:102:24
    |
102 |             Some(libc::SIGABRT) => TestResult::TrFailed,
    |                        ^^^^^^^ not found in `libc`
```

**rust-analyzer version**: 0.4.1838-standalone (c06ca6cff 2024-02-11) + 0.3.1839-standalone (c06ca6cff 2024-02-11)

**rustc version**: rustc 1.78.0-nightly (b381d3ab2 2024-02-12)

OS: Mac(M1)

fresh project generated via: `cargo generate esp-rs/esp-idf-template cargo`

cargo build & check works fine and don't throw this error, only r-a
if I remove the line in `test_result.rs` everything works fine and r-a finishes without an error

if you need further information or if I can do anything to narrow the scope down please just tell

**Comment by lnicola** [comment]
That's a file from the standard library, why would you edit it?

**Comment by Zercerium** [comment]
nah i don't want to edit it 😅. Was just for testing, if the error is then gone.

**Comment by lnicola** [comment]
So what error are you getting when you don't have the file open?

**Comment by Zercerium** [comment]
```rs
2024-02-13T17:35:30.838956Z ERROR rust_analyzer::main_loop: FetchBuildDataError:
error[E0531]: cannot find unit struct, unit variant or constant `SIGABRT` in crate `libc`
   --> /Users/nick/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/test/src/test_result.rs:102:24
    |
102 |             Some(libc::SIGABRT) => TestResult::TrFailed,
    |                        ^^^^^^^ not found in `libc`


error: aborting due to 1 previous error


For more information about this error, try `rustc --explain E0531`.

error: could not compile `test` (lib) due to 2 previous errors
```

I don't have this file normally open, this error shows up in the r-a: LS window after creating a new project with the command `cargo generate esp-rs/esp-idf-template cargo` and after r-a finishes with it run

**Comment by gvozdvmozgu** [comment]
if you add a config.toml file to the project with
```
[build]
rustflags = ["-A", "warnings"]
```
![image](https://github.com/rust-lang/rust-analyzer/assets/131762131/93eea506-672b-4e71-96d7-a66d928fca3b)
then many errors appear in other people's crates, although I haven't even opened them

**Comment by Kezii** [comment]
I'm having the same issue, rust analyzer breaks when using an esp-idf project

```
2024-02-15T21:41:26.933826Z ERROR rust_analyzer::main_loop: FetchBuildDataError:
error[E0531]: cannot find unit struct, unit variant or constant `SIGABRT` in crate `libc`
   --> /home/[...]/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/test/src/test_result.rs:102:24
    |
102 |             Some(libc::SIGABRT) => TestResult::TrFailed,
    |                        ^^^^^^^ not found in `libc`


error: aborting due to 1 previous error


For more information about this error, try `rustc --explain E0531`.

error: could not compile `test` (lib) due to 2 previous errors
```

**Comment by mrchantey** [comment]
Yes same got the error on an `esp-idf` project after upgrading to 
```
rustc 1.78.0-nightly (ee9c7c940 2024-02-14)
```

**Comment by Veykril** [comment]
Does this happen with the latest nightly rust-analyzer build? 2024-02-15 / 2024-02-16

**Comment by Zercerium** [comment]
> Does this happen with the latest nightly rust-analyzer build? 2024-02-15 / 2024-02-16

Yes, updates to rust-analyzer version: 0.4.1843-standalone (b9b0d29b8 2024-02-15) and rustc 1.78.0-nightly (a4472498d 2024-02-15)

**Comment by weiying-chen** [comment]
I'm having the [same](https://github.com/esp-rs/esp-idf-svc/issues/366) issue with the same error.

**Comment by weiying-chen** [comment]
> Yes same got the error on an `esp-idf` project after updating rust nightly to
> 
> ```
> rustc 1.78.0-nightly (ee9c7c940 2024-02-14)
> ```

@mrchantey What rust nightly version doesn't have this error?

**Comment by mrchantey** [comment]
> @mrchantey Which rust nightly version doesn't have this error?

Pretty sure I was on the previous version `1.77.0-nightly`

**Comment by weiying-chen** [comment]
@mrchantey I confirmed it: rust-analyzer in rustc 1.77.0-nightly (2a3e63551 2023-12-30) works perfectly fine in esp-idf procjets.

**Comment by Zercerium** [comment]
seems plausible cause the line was added here, if I'm correct https://github.com/rust-lang/rust/commit/f622e832d41171578b6987d0b25b3a84d6d1c81c#diff-c91a894be23c4fb9ac51642c91c20500cdd48f7f0114b4b3c6a8a1cb47afa674

**Comment by jellyterra** [comment]
![1711989266791](https://github.com/rust-lang/libc/assets/80201595/1a856fe2-77f1-4380-9d98-eab2a721dee6)
I have similar problem on amd64

**Comment by milewski** [comment]
Same problem here, temporarily I'm commenting out that offending line and it works:

```rust
None => match status.signal() {
    // Some(libc::SIGABRT) => TestResult::TrFailed,
    Some(signal) => {
        TestResult::TrFailedMsg(format!("child process exited with signal {signal}"))
    }
    None => unreachable!("status.code() returned None but status.signal() was None"),
},
```

**Comment by Veykril** [comment]
Try setting `"rust-analyzer.cargo.allTargets": false,` (needs latest nightly r-a, from 01.04.2024) if you are working on a no-std target. If you still run into that with that setting we'll need more info.

**Comment by Tevz-Beskovnik** [comment]
Enabling that setting does not resolve the issue.

**Comment by faulesocke** [comment]
Also getting that very same error when running `cargo check --tests` for my esp-idf projects. Seems like the error really is in the `test` library (or one of it's dependencies).

**Comment by faulesocke** [comment]
When I set the option `cargo.allTargets` to `false`, this makes it work without patching the source. Indeed, passing `--all-targets` seems to have the same effect as passing `--tests` to `cargo check`, at least in regards to this bug.

Doing this in my neovim+lazy.nvim setup works (for example) like this:
```lua
{
  "mrcjkb/rustaceanvim",
  opts = function(_, opts)
    opts.server.default_settings["rust-analyzer"].cargo.allTargets = false
  end,
}
```

But note that this is only a workaround.

**Comment by Veykril** [comment]
Okay thanks for confirming that the setting works, then I'll close this in favor of https://github.com/rust-lang/rust-analyzer/issues/14205
